src/components/editor/ImportResourceTemplate.jsx

```diff
@@ -43,80 +44,102 @@ class ImportResourceTemplate extends Component {
-      this.setState({tempConflictError: response})
-      joinedConflicts.push(this.state.tempConflictError)
-      this.setState({createResourceError: joinedConflicts})
+  updateResource = async (content, group) => {
+    let response
+    try {
+      response = await instance.updateResourceWithHttpInfo(group, content.id, content, { contentType: 'application/json' })
+    } catch(error) {
+      console.log('received error with update resource request:')
+      console.dir(error)
+      this.setState({ updateResourceError: error })
+      return error.response
     }
+    console.log('received non-error response to update resource request:')
+    console.dir(response)
+    return response.response
   }
 
-  updateAllResources = (rts, group) => {
-    return Promise.all(rts.map(async rt =>
-      await this.fulfillCreateResourcePromise(this.updateResourcePromise(rt, group))
-    ))
+  handleResponse = (response, operation, profileCount) => {
+    console.log(`handling ${operation} response:`)
+    console.dir(response)
+    // HTTP status 409 == Conflict
+    if (operation == 'update' && response.status !== 409) {
+      const msg = 'The sinopia server is not accepting the request for this resource.'
+      this.state.updateResourceError ? this.setState({message: [`${msg}: ${this.state.updateResourceError}`]}) : this.setState({message: [msg]})
+    }
+
+    // HTTP status 401 == Unauthorized
+    if (response.status === 401) {
+      return
+    }
+
+    console.log(`checking if response is conflict (${response.status})`)
+    console.log(`checking if CRE.length (${this.state.createResourceError.length}) == profileCount (${profileCount})`)
+    // HTTP status 409 == Conflict
+    if (response.status === 409 && this.state.createResourceError.length >= profileCount) {
+      this.setState({ modalShow: true })
+    }
+
+    const joinedMessages = this.state.message.slice(0)
+    console.log('pushing to messages (.status and .headers.location):')
+    console.dir(response)
+    const location = response.headers.location || ''
+    this.setState({ tempState: `${this.humanReadableStatus(response.status)} ${location}` })
+    joinedMessages.push(this.state.tempState)
+    this.setState({ message: joinedMessages })
+  }
+
+  humanReadableStatus = status => {
+    switch(status) {
+      case 201:
+        return 'Created'
+      case 204:
+        return 'Updated'
+      case 401:
+        return 'You are not authorized to do this. Try logging in again!'
+      case 409:
+        return 'Attempting to update'
+      default:
+        return `Unexpected response (${status})!`
+    }
   }
 
   handleUpdateResource = async (rts, group) => {
-    await this.updateAllResources(rts, group).then(() => {
-      const incrementedKey = this.state.updateKey + 1
-      this.setState({updateKey: incrementedKey})
-      this.modalClose()
-      window.location.reload()
+    rts.forEach(async rt => {
```

Comment:
would there be any gain to using Promise.all to wait for all to return here? (prob not)

Reply:
I'm trying to avoid overt promises wherever possible. Prolly worth broader team discussion.

Reply:
Since you are using the anonymous arrow function in this `forEach`, you could use the `map` operator here as well.